bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/skeleton/factory/SpProgramFactory.scala

```diff
@@ -201,11 +201,16 @@ object SpProgramFactory {
       timeAccountingRatios(proposal) match {
         case Nil    => None
         case ratios =>
+          val progTime = hoursFromObservations(proposal, _.progTime)
+          val partTime = hoursFromObservations(proposal, _.partTime)
+          require((hrs - (progTime |+| partTime).hours).abs < 1e-1,
+            "Awarded time must be equal to the sum of program time and partner calibration time")
+
           val jmap = ratios.map { case (cat, rat) =>
-            // REL-2928 TODO: handle splitting program and partner award.  Here
-            // we just keep everything as the program award for now.
-            val ms    = ((hrs * rat) * MsPerHour).round // ms for this category
-            val award = new TimeAcctAward(Duration.ofMillis(ms), Duration.ZERO)
+            def duration(v: TimeAmount): Duration =
+              Duration.ofMillis(((v.hours * rat) * MsPerHour).round)
```

Comment:
I'm not sure of a way offhand, as the `java.time.Duration.ofXXX` always requires a `long` and we are working with `double` values in `TimeAmount`.